src/use-query.ts

```diff
@@ -73,6 +73,15 @@ export function useQuery<TResult, TError = ErrorDefault>(
   const refresh = () => store.refresh(entry.value)
   const refetch = () => store.refetch(entry.value)
 
+  // if passed by client
+  if (_options.enabled !== undefined) {
+    watch(options.enabled, (newEnabled, oldEnabled) => {
+      // add test case why we check !oldEnabled
+      // add test case when we update entry, add test case when we update enabled
+      if (!oldEnabled && newEnabled) refetch()
```

Comment:
```suggestion
      if (!oldEnabled && newEnabled) refresh()
```

It should always be refresh by default. Maybe it's worth adding another unit test that ensures the query function is not called when enabled goes from true to false to true within the staletime